### PR TITLE
Upgrade aws

### PR DESCRIPTION
**Pull request by skymandr**
This upgrades the dependencies and, most crucially, updates to using v2 of the AWS cli. We should probably make a release once this and #135 are landed.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -3,13 +3,31 @@ module github.com/innosat-mats/rac-extract-payload
 go 1.19
 
 require (
-	github.com/aws/aws-sdk-go v1.31.0
+	github.com/aws/aws-sdk-go-v2 v1.16.16
+	github.com/aws/aws-sdk-go-v2/config v1.17.8
+	github.com/aws/aws-sdk-go-v2/feature/s3/manager v1.11.34
+	github.com/aws/aws-sdk-go-v2/service/s3 v1.27.11
```
Comment (noreklint) [inline go.mod:9]: I know in Typescript there's an issue if all internal cdk (not sure if it applies to sdk as well) dependencies aren't of the same version. These seem to be the latest version, so probably all good?

File: internal/awstools/uploader.go [hunk 2]
```diff
@@ -1,28 +1,29 @@
...
+	"github.com/aws/aws-sdk-go-v2/service/s3"
 )
 
 const awsBucket = "mats-l0-artifacts"
 
 // AWSUpload uploads file content to target bucket
-func AWSUpload(uploader *s3manager.Uploader, key string, body io.Reader) {
-	_, err := uploader.Upload(&s3manager.UploadInput{
+func AWSUpload(uploader *manager.Uploader, key string, body io.Reader) {
+	_, err := uploader.Upload(context.TODO(), &s3.PutObjectInput{
```
Comment (noreklint) [inline internal/awstools/uploader.go:18]: What's this `context.TODO()`?

**Review by noreklint — CHANGES_REQUESTED** [review]
LCTM, but some questions

File: go.mod (hunk 1 above)
Comment (skymandr) [reply go.mod:9]: Unfortunately (or fortunately? maybe the go version is more sensible than npm..?) the go packages don't seem to follow the versioning of the actual SDK. I've tried it locally and it works...

File: internal/awstools/uploader.go (hunk 2 above)
Comment (skymandr) [reply internal/awstools/uploader.go:18]: Surprised me too, but it seems to be what they do in the examples: https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/feature/s3/manager#NewUploader (see also: https://asanchez.dev/blog/amazon-s3-v2-golang/ for a more rigorous example).

The function is a go standard library function, documented here: https://pkg.go.dev/context#TODO.

File: go.mod (hunk 1 above)
Comment (skymandr) [reply go.mod:9]: (And when I say locally, I mean that I have processed a number of rac files locally with the `-aws` option and checked that they showed up in the expected bucket.)

**Review by noreklint — APPROVED** [review]

**Pull request merged**